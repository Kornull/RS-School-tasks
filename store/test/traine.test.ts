import btnColors from '../src/components/viev/filter-blocks/filter-color';
import updateCards from '../src/components/controller/products/product';
import arr from '../src/laptop.json';

/**
 * @jest-environment jsdom
 */
const btnColor = new btnColors();
const updateLink = new updateCards();

const card = {
  brand: 'Asus',
  model: 'Asus TUF Gaming',
  number: 10,
  ram: '16',
  color: 'black',
  image: 'asus_tuf_gaming_f15_fx507zm-hn001.jpg',
  year: '2022',
  description: [
    {
      'Screen diagonal': '15.6',
    },
    {
      'Screen resolution': '1920x1080 (FullHD)',
    },

    {
      'Processor model': 'Intel Core i7-12700H',
    },
    {
      'The amount of RAM': '16 GB',
    },
    {
      SSD: '1024 GB',
    },
    {
      'Video card model': 'NVIDIA GeForce RTX 3060',
    },
    {
      'Operating system': 'DOS',
    },
    {
      Cost: '$2343.0',
    },
  ],
};

describe('View and create color btns are in the block', () => {
  it('Create btns', () => {
    expect(btnColor.filterColor()).toMatchSnapshot();
  });
  it('Add class to the block btns', () => {
    expect(btnColor.filterColor().classList.contains('filter__btn--color'));
  });
  it('Add class to btns', () => {
    expect(btnColor.filterColor().classList.contains('btn__color'));
  });
});

describe('Create cards', () => {
  it('Create cards', () => {
    expect(updateLink.create()).toMatchSnapshot();
  });
});

describe('Array length and keys are in object', () => {
  it('length array = 21', () => {
    expect(arr).toHaveLength(21);
  });
  it('There are keys in object', () => {
    expect(arr).toContainEqual(card);
  });
});
