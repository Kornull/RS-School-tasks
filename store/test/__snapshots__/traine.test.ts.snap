// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create cards Create cards 1`] = `
<div
  class="main__product"
>
  <div
    class="card"
    id="Lenovo-IdeaPad-3-15ITL05"
  >
    Lenovo IdeaPad 3 15ITL05
    <img
      class="card__img"
      src="./assets/img/lenovo_ideapad_3_15itl05.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Pentium Gold 7505
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - Intel UHD Graphics Xe G4
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $512.0
      </li>
    </ul>
    <div>
      In stock - 16
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Asus-TUF-Gaming"
  >
    Asus TUF Gaming
    <img
      class="card__img"
      src="./assets/img/asus_tuf_gaming_f15_fx507zm-hn001.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Core i7-12700H
      </li>
      <li>
        The amount of RAM - 16 GB
      </li>
      <li>
        SSD - 1024 GB
      </li>
      <li>
        Video card model - NVIDIA GeForce RTX 3060
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $2343.0
      </li>
    </ul>
    <div>
      In stock - 10
    </div>
    <div>
      Year - 2022
    </div>
  </div>
  <div
    class="card"
    id="Dell-Vostro-15"
  >
    Dell Vostro 15
    <img
      class="card__img"
      src="./assets/img/dell_vostro_15_3515-284698_42.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - AMD Ryzen 7 3700U
      </li>
      <li>
        RAM - 16 GB (1x16 GB)
      </li>
      <li>
        SSD - 512 GB
      </li>
      <li>
        Video card model - AMD Radeon Vega 10
      </li>
      <li>
        Operating system - Ubuntu
      </li>
      <li>
        Cost - $1135.0
      </li>
    </ul>
    <div>
      In stock - 4
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Lenovo-IdeaPad-5-15ITL05"
  >
    Lenovo IdeaPad 5 15ITL05
    <img
      class="card__img"
      src="./assets/img/lenovo_ideapad_5_15itl05.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Core i3-1115G4
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 512 GB
      </li>
      <li>
        Video card model - NVIDIA GeForce MX450
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $915.0
      </li>
    </ul>
    <div>
      In stock - 8
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Honor-MagicBook-16"
  >
    Honor MagicBook 16
    <img
      class="card__img"
      src="./assets/img/honor_magicbook_16_17.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 16.1
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - AMD Ryzen 5 5600H
      </li>
      <li>
        RAM - 16 GB
      </li>
      <li>
        SSD - 512 GB
      </li>
      <li>
        Video card model - AMD Radeon Vega 7
      </li>
      <li>
        Operating system - Windows 11 Home
      </li>
      <li>
        Cost - $1208.0
      </li>
    </ul>
    <div>
      In stock - 4
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Lenovo-IdeaPad-5-Pro"
  >
    Lenovo IdeaPad 5 Pro
    <img
      class="card__img"
      src="./assets/img/lenovo_ideapad_5_pro_16ach6.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 16.0
      </li>
      <li>
        Screen resolution - 2560x1600 (WQXGA)
      </li>
      <li>
        Processor model - AMD Ryzen 5 5600H
      </li>
      <li>
        The amount of RAM - 16 GB
      </li>
      <li>
        SSD - 512 GB (M.2 2280 PCIe 3.0x4 NVMe)
      </li>
      <li>
        Video card model - AMD Radeon Vega 7
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $1208.0
      </li>
    </ul>
    <div>
      In stock - 15
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Acer-Aspire-3."
  >
    Acer Aspire 3.
    <img
      class="card__img"
      src="./assets/img/acer_aspire_3_a315-34-c786.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Celeron N4020
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - Intel UHD Graphics 600
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $439.0
      </li>
    </ul>
    <div>
      In stock - 14
    </div>
    <div>
      Year - 2020
    </div>
  </div>
  <div
    class="card"
    id="Honor-MagicBook-15-2021"
  >
    Honor MagicBook 15 2021
    <img
      class="card__img"
      src="./assets/img/honor_magicbook_15_2021_22.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - AMD Ryzen 5 5500U
      </li>
      <li>
        RAM - 8 GB
      </li>
      <li>
        SSD - 512 GB
      </li>
      <li>
        Video card model - AMD Radeon Vega 7
      </li>
      <li>
        Operating system - Windows 10
      </li>
      <li>
        Cost - $952.0
      </li>
    </ul>
    <div>
      In stock - 6
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="HP-250-G8"
  >
    HP 250 G8
    <img
      class="card__img"
      src="./assets/img/hp_250_g8.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Core i5-1135G7
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 512 GB
      </li>
      <li>
        Video card model - Iris Xe Graphics G7
      </li>
      <li>
        Operating system - Windows 10
      </li>
      <li>
        Cost - $842.0
      </li>
    </ul>
    <div>
      In stock - 4
    </div>
    <div>
      Year - 2020
    </div>
  </div>
  <div
    class="card"
    id="Lenovo-IdeaPad-Gaming-3"
  >
    Lenovo IdeaPad Gaming 3
    <img
      class="card__img"
      src="./assets/img/lenovo_ideapad_gaming_3_57.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - AMD Ryzen 5 5600H
      </li>
      <li>
        RAM - 16 GB (1x 16 GB)
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        HDD - 1000 GB
      </li>
      <li>
        Video card model - NVIDIA GeForce RTX 3050
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $1354.0
      </li>
    </ul>
    <div>
      In stock - 2
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Asus-ROG-Strix-G15"
  >
    Asus ROG Strix G15
    <img
      class="card__img"
      src="./assets/img/asus_rog_strix_g15_g513ih-hn014_16.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - AMD Ryzen 7 4800H
      </li>
      <li>
        RAM - 16 GB
      </li>
      <li>
        SSD - 512 GB
      </li>
      <li>
        Video card model - NVIDIA GeForce GTX 1650
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $1538.0
      </li>
    </ul>
    <div>
      In stock - 5
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Acer-Aspire-3"
  >
    Acer Aspire 3
    <img
      class="card__img"
      src="./assets/img/acer_aspire_3_a315-34-p0x8.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Pentium Silver N5030
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - Intel UHD Graphics 600
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $512.0
      </li>
    </ul>
    <div>
      In stock - 7
    </div>
    <div>
      Year - 2020
    </div>
  </div>
  <div
    class="card"
    id="Asus-VivoBook"
  >
    Asus VivoBook
    <img
      class="card__img"
      src="./assets/img/asus_vivobook_x543ma-dm1370.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Celeron N4020
      </li>
      <li>
        The amount of RAM - 4 GB
      </li>
      <li>
        HDD - 1 TB
      </li>
      <li>
        Video card model - Intel UHD Graphics 600
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $402.0
      </li>
    </ul>
    <div>
      In stock - 9
    </div>
    <div>
      Year - 2019
    </div>
  </div>
  <div
    class="card"
    id="Lenovo-IdeaPad-L3."
  >
    Lenovo IdeaPad L3.
    <img
      class="card__img"
      src="./assets/img/lenovo_ideapad_l3_15itl6.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Pentium Gold 7505
      </li>
      <li>
        The amount of RAM - 8 GB 
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - Intel UHD Graphics
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $512.0
      </li>
    </ul>
    <div>
      In stock - 13
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Asus-VivoBook-Flip-14"
  >
    Asus VivoBook Flip 14
    <img
      class="card__img"
      src="./assets/img/asus_vivobook_flip_14_tp1401ka-bz063.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 14.0
      </li>
      <li>
        Screen resolution - 1366x768 (HD)
      </li>
      <li>
        Processor model - Intel Pentium Silver N6000
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - Intel UHD Graphics
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $622.0
      </li>
    </ul>
    <div>
      In stock - 8
    </div>
    <div>
      Year - 2022
    </div>
  </div>
  <div
    class="card"
    id="HP-255-G8"
  >
    HP 255 G8
    <img
      class="card__img"
      src="./assets/img/hp_255_g8.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - AMD Ryzen 3 5300U
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - AMD Radeon Vega 6
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $805.0
      </li>
    </ul>
    <div>
      In stock - 14
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Honor-MagicBook-X15"
  >
    Honor MagicBook X15
    <img
      class="card__img"
      src="./assets/img/honor_magicbook_x15.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Core i5-10210U
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 512 GB
      </li>
      <li>
        Video card model - Intel UHD Graphics 620
      </li>
      <li>
        Operating system - Windows 10
      </li>
      <li>
        Cost - $1061.0
      </li>
    </ul>
    <div>
      In stock - 8
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Lenovo-IdeaPad-L3"
  >
    Lenovo IdeaPad L3
    <img
      class="card__img"
      src="./assets/img/lenovo_ideapad_l3_15itl6.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - Intel Celeron 6305
      </li>
      <li>
        The amount of RAM - 4 GB
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - Intel UHD Graphics
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $549.0
      </li>
    </ul>
    <div>
      In stock - 9
    </div>
    <div>
      Year - 2021
    </div>
  </div>
  <div
    class="card"
    id="Apple-MacBook-Air-13"
  >
    Apple MacBook Air 13
    <img
      class="card__img"
      src="./assets/img/apple_macbook_air_13.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 13.3
      </li>
      <li>
        Screen resolution - 2560x1600 (Retina 13)
      </li>
      <li>
        Processor model - Apple M1
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - Apple M1 GPU
      </li>
      <li>
        Operating system - macOS
      </li>
      <li>
        Cost - $1904.0
      </li>
    </ul>
    <div>
      In stock - 2
    </div>
    <div>
      Year - 2020
    </div>
  </div>
  <div
    class="card"
    id="Lenovo-V15-ADA"
  >
    Lenovo V15-ADA
    <img
      class="card__img"
      src="./assets/img/lenovo_v15-ada.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - AMD Ryzen 5 3500U
      </li>
      <li>
        The amount of RAM - 8 GB
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - AMD Radeon Vega 8
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $732.0
      </li>
    </ul>
    <div>
      In stock - 7
    </div>
    <div>
      Year - 2020
    </div>
  </div>
  <div
    class="card"
    id="HP-15s-eq1328ur"
  >
    HP 15s-eq1328ur
    <img
      class="card__img"
      src="./assets/img/hp_15s-eq1328ur.jpg"
    />
    <ul
      class="card__list"
    >
      <li>
        Screen diagonal - 15.6
      </li>
      <li>
        Screen resolution - 1920x1080 (FullHD)
      </li>
      <li>
        Processor model - AMD Athlon Silver 3050U
      </li>
      <li>
        The amount of RAM - 8 GB (1x8GB)
      </li>
      <li>
        SSD - 256 GB
      </li>
      <li>
        Video card model - AMD Radeon Vega 2
      </li>
      <li>
        Operating system - DOS
      </li>
      <li>
        Cost - $512.0
      </li>
    </ul>
    <div>
      In stock - 11
    </div>
    <div>
      Year - 2020
    </div>
  </div>
</div>
`;

exports[`View and create color btns are in the block Create btns 1`] = `
<div
  class="filter__color"
>
  <div
    class="filter__btn--color"
  >
    <div />
    <button
      class="btn btn__color"
      id="silver"
      style="background-color: silver;"
    />
    <button
      class="btn btn__color"
      id="black"
      style="background-color: black;"
    />
    <button
      class="btn btn__color"
      id="gray"
      style="background-color: gray;"
    />
    <button
      class="btn btn__color"
      id="PeachPuff"
      style="background-color: PeachPuff;"
    />
  </div>
</div>
`;
